### PR TITLE
mac: Distinguish trackpad from Magic Mouse scroll

### PR DESCRIPTION
**Pull request by nanoant**
This is the second attempt to distinguish trackpad scrolling that when used
should yield panning from Magic Mouse scrolling that should control zoom.

Since trackpad will begin with a touch prior to the scroll event, but Magic
Mouse not, we can use it to make sure we trigger panning only on a trackpad.

Previous "mac: Don't interpret single-touch scroll events as pan gestures"
that was flawed was reverted which ultimately lead to being unable to zoom using
Magic Mouse.

This fixes again #1107 and is also related to #1218

### CODE REVIEW

**Comment by ruevs** [comment]
Just for cross-reference - the original issue is here https://github.com/solvespace/solvespace/issues/1106

**Comment by nanoant** [comment]
@vespakoen Thanks for the report. I managed to reproduce and then fix the problem. Can you pull the latest update.

TL;DR Actually we can have multiple touches when we accidentally touch trackpad with another finger when scrolling fast. Using counter instead of simple flag seems to resolve the problem to me and I am no longer able to trigger zoom with scrolling widely on my MBP with 2 fingers, while Magic Mouse on my Mini works great as well.

**Comment by vespakoen** [comment]
This is working for me now as well.

Thanks for working on this and figuring out a way to distinguish between mice ;)

Merging...
